src/core/signal.cc

```diff
@@ -0,0 +1,50 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifdef SEASTAR_MODULE
+module;
+#endif
+
+#include <stdexcept>
+
+#ifdef SEASTAR_MODULE
+module seastar;
+#else
+#include <seastar/core/signal.hh>
+#include <seastar/core/reactor.hh>
+#endif
+
+namespace seastar {
+
+static reactor& engine_or_throw() {
+    if (!engine_is_ready()) {
```

Comment:
I think that's impossible. No tasks can be run until engine is ready

Reply:
Its possible, if a user calls the procedure before `app.run()`, like:

```C++
seastar::handle_signal(SIGINT, [] {});
return app.run(argc, argv, [&app] -> seastar::future<> {
...
```

I dont think its obvious that it must be called inside `run()`, so I think the logic provides a more controlled and understandable error handling.

Reply:
In this case the "reactor must be initialized" can be confusing message, because it's not obvious that "app.run()" initializes the reactor. I would just mention in the method doc, that it's only valid to use one from inside app.run() lambda, otherwise it's UB